### PR TITLE
HDDS-2875. Add a config in ozone to tune max outstanding requests in …

### PR DESCRIPTION
**Pull request by bharatviswa504**
…raft client.

## What changes were proposed in this pull request?

Add a config to tune the config value of "raft.client.async.outstanding-requests.max" config in raft client. 

There is a property scm.container.client.max.outstanding.requests but this is used to set outStandingAppendsMax, but the config of this is mentioned as the "Controls the maximum number of outstanding async requests that can be handled by the Standalone as well as Ratis client". So, I removed this from ScmClientConfig, and created a new class for this property with prefix dfs.ratis.client to be consistent with other ratis client properties. And used the same for outStandingAppendsMax as before, not sure this is the correct way, but to keep same with current code, used it.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-2875

Please replace this section with the link to the Apache JIRA)

## How was this patch tested?

UT and acceptance test run.

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java [hunk 1]
```diff
@@ -206,6 +206,10 @@ static RaftClient newRaftClient(RpcType rpcType, RaftPeerId leader,
     GrpcConfigKeys.OutputStream.setOutstandingAppendsMax(properties,
```
Comment (lokeshj1703) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java]: I think we can remove this setter. This is used by GrpcOutputStream in ratis which is not used by ozone.

File: hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientManager.java [hunk 2]
```diff
@@ -345,10 +337,30 @@ public void setMaxSize(int maxSize) {
...
 
+  }
+
+  /**
+   * Configuration for ratis client.
+   */
+  @ConfigGroup(prefix = "dfs.ratis.client")
+  public class DFSRatisClientConfig {
+
+    @Config(key = "async.max.outstanding.requests",
```
Comment (lokeshj1703) [inline hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientManager.java]: I dont think we have a corresponding config in ozone-default.xml.

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java (hunk 1 above)
Comment (bharatviswa504) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java]: Done

File: hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientManager.java (hunk 2 above)
Comment (bharatviswa504) [reply hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientManager.java]: With new-style Java config, we don't need to add, it generates an ozone-default-generated.xml. In that file, I can see this new config.

```
[root@ozone-test-1 ex]# cat ozone-default-generated.xml | grep -A4 "dfs.ratis.client.async"
    <name>dfs.ratis.client.async.max.outstanding.requests</name>
    <value>64</value>
    <description>Controls the maximum number of outstanding async requests that can be handled by the Standalone as well as Ratis client.</description>
    <tag>OZONE, CLIENT, PERFORMANCE</tag>
  </property>
```

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java (hunk 1 above)
Comment (bharatviswa504) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java]: Done

**Comment by bharatviswa504** [comment]
@lokeshj1703 Thank You for the review. Addressed review comments and acceptance test failures.

**Comment by lokeshj1703** [comment]
@bharatviswa504 Thanks for updating the PR! The changes look good to me. I have committed it to master branch.

**Pull request merged**